clarify modeling layer assumptions in README (#264)
dbt-labs/dbt-project-evaluator: README.md

2. Query the associated model to find the specific instances of the issue within your project
3. Either fix the issue(s) or [customize](#customization) the package to exclude them

#### Caveat Regarding Modeling Layer Assumptions

> This package assumes that you are using standard modeling layers (ie staging, intermediate, and marts) and that those layers are reflected in both the folder structure and naming prefixes used in your project. If you utilize these modeling layers but have prefixes other than `stg_`, `int_`, `fct_`, and `dim_`, you may override which prefixes the package should look for (see [Overriding Variables](#overriding-variables)). If your project doesn't use these modeling layers, you may wish to ignore or disable package models that don't apply in your case (ie [Downstream Models Dependent on Source](#downstream-models-dependent-on-source)).

Reviewer: I feel like this is more of a default rather than an assumption? 

And I would avoid listing out the default prefixes here since they're already captured in the Overriding Variables section.

Author: It's definitely a default, and I would argue it's an assumption since there isn't really any way to configure or use these five models if you don't use modeling layers of some kind. 

Regarding removing the default prefixes, I see your point from an updatability perspective, but I also think these values are unlikely to change (and certainly not frequently). It feels like a trade-off in usability (ie requiring a second click if you got unexpected results from your model), and I'd be inclined to leave it as is. How strongly do you feel about this?

Reviewer: I'm going to add a comment with suggested changes so we can see how it would read

Reviewer: Just want to avoid repeating code since we already have a very similar paragraph:

> The model_types, <model_type>_folder_name, and <model_type>_prefixes variables allow the package to check if models in the different layers are in the correct folders and have a correct prefix in their name. The default model types are the ones we recommend in our [dbt Labs Style Guide](https://github.com/dbt-labs/corp/blob/main/dbt_style_guide.md). If your model types are different, you can update the model_types variable and create new variables for <model_type>_folder_name and/or <model_type>_prefixes.

Reviewer: How would you feel about just linking to this section instead for each of the relevant models?

Author: Oh, good call - totally down for this! I've updated the links accordingly (not 100% sure the anchor link is working as expected, since that section isn't a header).

I'm wondering if we just ditch the other paragraph I had created then? Unless we want to call this out explicitly early in the readme.

> This package assumes that you are using standard modeling layers (ie staging, intermediate, and marts) and that those layers are reflected in both the folder structure and naming prefixes used in your project. If you utilize these modeling layers but have prefixes other than `stg_`, `int_`, `fct_`, and `dim_`, you may override which prefixes the package should look for (see Naming Convention Variables in [Overriding Variables](#overriding-variables)). If your project doesn't use these modeling layers, you may wish to ignore or disable package models that don't apply in your case (ie [Downstream Models Dependent on Source](#downstream-models-dependent-on-source)).

Author: This definitely feels like an improvement - thanks! Preference for keeping this or just getting rid of it since it's so similar to the Naming Convention Variable section?

`fct_marts_or_intermediate_dependent_on_source` ([source](models/marts/dag/fct_marts_or_intermediate_dependent_on_source.sql)) shows each downstream model (`marts` or `intermediate`)
that depends directly on a source node.

> Note: if your naming conventions don't **correlate to modeling layer** (ie `stg`, `int`, `fct`, etc. or another custom naming pattern defined in the [Overriding Variables](#overriding-variables) section below), this check will return no results.

Reviewer: ```suggestion
Note: if your naming conventions don't **correlate to relevant modeling layers** (i.e. `marts` and `intermediate`), this check will return no results.
```
I think this is currently hard-coded

`fct_marts_or_intermediate_dependent_on_source` ([source](models/marts/dag/fct_marts_or_intermediate_dependent_on_source.sql)) shows each downstream model (`marts` or `intermediate`)
that depends directly on a source node.

> If you get no results or unexpected results, see [Modeling Layer Assumptions](#caveat-regarding-modeling-layer-assumptions) for more context.

Author: Truth! Appreciate the feedback - updates will be committed momentarily!
